src/main/java/edu/oakland/production/display01/DisplayGpsInterfaceImplementation.java

```diff
@@ -1,23 +1,67 @@
 package edu.oakland.production.display01;
 
 import edu.oakland.helper.admin.Satellite;
+import edu.oakland.helper.display01.SatelliteSignalCheckRequest;
+import edu.oakland.production.display01.DisplayGpsReceiver;
 
 public class DisplayGpsInterfaceImplementation implements DisplayGpsInterface {
+  
+  public DisplayGpsReceiver reciever;
 
-  public boolean receiveGpsSignal(Satellite satelliteSignal) {
-    return true;
+  public DisplayGpsInterfaceImplementation(DisplayGpsReceiver reciever) {
+    this.reciever = reciever;
   }
 
-  public String reportGpsSignalLoss(Satellite satelliteSignal) {
-    return "";
+  /** 
+   * Recieves GPS Signal Strength.
+   *
+   * @param satelliteSignal GPS signal to measure.
+   * @return Which GPS Recived 
+   */
+  public String receiveGpsSignal(Satellite satelliteSignal) {
+    System.out.println("Recieved GPS Signal from:");
+    String satellite = reciever.measureGpsSignalStrength(satelliteSignal);
+    System.out.println(satellite);
+    return satellite;
   }
 
-  public int recheckSignalStrength(Satellite satelliteSignal) {
-    return 0;
+  /** 
+   * Reports GPS Signal Loss.
+   *
+   * @param satelliteSignal GPS signal to report.
+   * @return Which GPS has Loss of Signal 
+   */
+  public SatelliteSignalCheckRequest reportGpsSignalLoss(Satellite satelliteSignal) {
+    System.out.println("Reporting loss for:");
+    SatelliteSignalCheckRequest satellite = reciever.measureSignal(satelliteSignal);
+    System.out.println(satellite.getSatelliteName());
```

Comment:
This is actually the next satellite to check, so this message should probably be something like:
```suggestion
    System.out.println("Attempting to " + satellite.getCheckType() + " " + satellite.getSatelliteName());
```